c7n/resources/vpc.py

```diff
@@ -114,101 +143,131 @@ class FlowLogFilter(Filter):
+            data['count_op'] = 'gte'
+            data['count'] = 1
+        else:
+            data['count'] = 0
+        attrs = []
+        for k in self.compat_conversion:
+            if k not in self.data:
                 continue
-            flogs = resource_map[r[m.id]]
-            r['c7n:flow-logs'] = flogs
-
-            # config comparisons are pointless if we only want vpcs with no flow logs
-            if enabled:
-                fl_matches = []
-                for fl in flogs:
-                    dest_type_match = (destination_type is None) or op(
-                        fl['LogDestinationType'], destination_type)
-                    if 'LogDestination' not in fl:
-                        fl['LogDestination'] = ''
-                    dest_match = (destination is None) or op(
-                        fl['LogDestination'], destination)
-                    status_match = (status is None) or op(fl['FlowLogStatus'], status.upper())
-                    delivery_status_match = (delivery_status is None) or op(
-                        fl['DeliverLogsStatus'], delivery_status.upper())
-                    traffic_type_match = (
-                        traffic_type is None) or op(
-                        fl['TrafficType'],
-                        traffic_type.upper())
-                    log_group_match = (log_group is None) or op(fl.get('LogGroupName'), log_group)
-                    log_format_match = (log_format is None) or op(fl.get('LogFormat'), log_format)
-                    # combine all conditions to check if flow log matches the spec
-                    fl_match = (status_match and traffic_type_match and dest_match and
-                                log_format_match and log_group_match and
-                                dest_type_match and delivery_status_match)
-                    fl_matches.append(fl_match)
-
-                if set_op == 'or':
-                    if any(fl_matches):
-                        results.append(r)
-                elif set_op == 'and':
-                    if all(fl_matches):
-                        results.append(r)
+            afilter = {}
+            cinfo = self.compat_conversion[k]
+            ak = cinfo['key']
+            av = self.data[k]
+            if 'values' in cinfo:
+                av = cinfo['values'][av]
+            if 'op' in self.data and self.data['op'] == 'not-equal':
+                av = {'value': av, 'op': 'not-equal'}
+            afilter[ak] = av
+            attrs.append(afilter)
+        if attrs:
+            data['attrs'] = attrs
+        data['type'] = self.type
+        self.source_data = self.data
+        self.data = data
+
+    def get_item_values(self, resource):
+        flogs = self.flow_log_map.get(resource[self.manager.resource_type.id], ())
+        # compatibility with v1 filter, we also add list-item annotation
+        # for matched flow logs
+        resource['c7n:flow-logs'] = flogs
```

Comment:
Looks like because of the policy I was using to test which is a legacy format policy as shown below. Since, I don't have `enabled: true` in the policy, the corresponding new format sets count : 0 rendering different results
```
policies:
  - name: vpc-flow-logs
    resource: vpc
    filters:
      - type: flow-logs
        status: active
        destination-type: s3
 ```
 ```
 {'count': 0, 'attrs': [{'FlowLogStatus': 'ACTIVE'}, {'LogDestinationType': 's3'}], 'type': 'flow-logs'}
 ```